test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs

```diff
@@ -603,6 +604,131 @@ public void TryGetPayloadValue(GetPayloadValueTheoryData theoryData)
             TestUtilities.AssertFailIfErrors(context);
         }
 
+        [Theory, MemberData(nameof(GetPayloadSubClaimValueTheoryData), DisableDiscoveryEnumeration = true)]
+        public void GetPayloadSubClaimValue(GetPayloadValueTheoryData theoryData)
+        {
+            CompareContext context = TestUtilities.WriteHeader($"{this}.GetPayloadSubClaimValue", theoryData);
+            try
+            {
+                JsonWebToken jsonWebToken = new JsonWebToken(theoryData.Json);
+                string payload = Base64UrlEncoder.Decode(jsonWebToken.EncodedPayload);
+                MethodInfo method = typeof(JsonWebToken).GetMethod("GetPayloadValue");
```

Comment:
we shouldn't need to use reflection....

Reply:
Yes but, it is more consistent with the rest of tests in this class to write it this way. What do you think?

Reply:
@keegan-caruso this is a way i found to call generic methods specifying [T]. There may be other way.

Reply:
Would have to frame it slightly differently. Could have had a func<object> on the theorydata that would retrieve the value instead of a type parameter. 

That would be a medium refactor. Lets consider it for later. Resolving for now.